tsinfer/formats.py

```diff
@@ -1597,24 +1585,13 @@ def add_site(
             raise ValueError(
                 "Site positions must be unique and added in increasing order"
             )
-
-        counts = allele_counts(genotypes)
-        if n_alleles > 2:
-            if inference is None:
-                inference = False
-            if inference:
-                raise ValueError("Only biallelic sites supported for inference")
-            if n_alleles > 64:
-                # This is mandated by tskit's map_mutations function.
-                raise ValueError("Cannot have more than 64 alleles")
-        if counts.derived > 1 and counts.derived < counts.known:
-            if inference is None:
-                inference = True
+        if inference is None:
```

Comment:
Do we want to entirely remove the concept of setting inference sites when we build the sample data file, and only specify them (if we want to override the default) when matching ancestors instead? We can work out from the ancestors_ts which sites were used for inference to create that file, so we wouldn't need to re-specify them when doing match_samples.

The way I might think of doing this is to provide an array of genome positions to match_ancestors, giving the sites that should *not* be used for inference. All others would use the tsinfer default.